pkg/compactor/block_upload.go

```diff
@@ -291,6 +300,8 @@ func (c *MultitenantCompactor) validateAndCompleteBlockUpload(logger log.Logger,
 			c.periodicValidationUpdater(ctx, logger, blockID, userBkt, cancel, validationHeartbeatInterval)
 		}()
 
+		c.blockUploadValidations.Inc()
+		defer c.blockUploadValidations.Dec()
```

Comment:
I suggest to move this at the beginning of this method, just after log statement.